src/Cake.Incubator/Project/ProjectParserExtensions.cs

```diff
@@ -31,10 +31,12 @@ namespace Cake.Incubator.Project
     [CakeAliasCategory("MSBuild Resource")]
     public static class ProjectParserExtensions
     {
-        private static readonly Regex NetCoreTargetFrameworkRegex = new Regex("([Nn][Ee][Tt])([Cc])\\w+", RegexOptions.Compiled);
-        private static readonly Regex NetStandardTargetFrameworkRegex = new Regex("([Nn][Ee][Tt])([Ss])\\w+", RegexOptions.Compiled);
-        private static readonly Regex NetFrameworkTargetFrameworkRegex = new Regex("([Nn][Ee][Tt][0-9*])\\w+", RegexOptions.Compiled);
-
+        private const RegexOptions parseOptions = RegexOptions.Compiled | RegexOptions.IgnoreCase;
+        private static readonly Regex NetCoreTargetFrameworkRegex = new Regex("^netcoreapp\\d+\\.\\d+$", parseOptions);
+        private static readonly Regex NetStandardTargetFrameworkRegex = new Regex("^netstandard\\d+\\.\\d+$", parseOptions);
+        private static readonly Regex NetFrameworkTargetFrameworkRegex = new Regex("^net\\d+$", parseOptions);
+        private static readonly Regex Net5PlusTargetFrameworkRegex = new Regex("^net\\d+\\.\\d+(-\\w+)?$", parseOptions);
```

Comment:
The target framework OS identifier may also include a version, example in the docs shows `net6.0-ios14.0` that wouldn't match `^net\d+\.\d+(-\w+)?$`. 

```suggestion
        private static readonly Regex Net5PlusTargetFrameworkRegex = new Regex("^net\\d+\\.\\d+(-[\\w.]+)?$", parseOptions);
```

https://docs.microsoft.com/en-us/dotnet/standard/frameworks#os-version-in-tfms

Reply:
Thanks @mryanmurphy, good catch.